### PR TITLE
feat: 🎸 get portfolio details for did

### PR DESCRIPTION
**Pull request by sansan**
### JIRA Link 

https://polymesh.atlassian.net/browse/DA-75

### Changelog / Description 

get portfolio details for did

### Checklist - 

- [x] New Feature ?
- [x] Updated swagger annotation (if API structure is changed) ?
- [x] Unit Test (if possible) ?
- [x] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/portfolios/portfolios.controller.ts [hunk 1]
```diff
@@ -143,4 +143,33 @@ export class PortfoliosController {
...
+    type: 'string',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
+  })
+  @ApiParam({
+    name: 'portfolioId',
+    description: 'The portfolioId for which details are to be fetched',
+    type: 'string',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
+  })
+  @ApiArrayResponse(PortfolioModel, {
```
Comment (polymath-eric) [inline src/portfolios/portfolios.controller.ts:163]: Isn't this finding just a single portfolio?

I don't think this should be an array response

File: src/portfolios/portfolios.controller.ts [hunk 2]
```diff
@@ -143,4 +143,33 @@ export class PortfoliosController {
...
+    description: 'Return the list of all Portfolios of the given Identity',
+    paginated: false,
+  })
+  @Get('/identities/:did/portfolios/:id')
+  async getPortfolio(@Param() { did, id }: PortfolioDto): Promise<PortfolioModel> {
+    const portfolio = await this.portfoliosService.findOne(did, id);
+
+    const result = await createPortfolioModel(portfolio, did);
```
Comment (polymath-eric) [inline src/portfolios/portfolios.controller.ts:171]:
```suggestion
    return createPortfolioModel(portfolio, did);
```

(nit) Nest will `await` the promise for you

**Review by polymath-eric — CHANGES_REQUESTED** [review]

**Comment by polymath-eric** [comment]
Also I think we want the branch to be merged into alpha so we don't have to do a `master` -> `alpha` merge.

I think the idea is to build up changes in alpha, but for minor version bumps I can see why its nice to merge right into master since it should be safe.

File: src/portfolios/portfolios.controller.ts [hunk 3]
```diff
@@ -143,4 +143,33 @@ export class PortfoliosController {
...
+  })
+  @ApiParam({
+    name: 'did',
+    description: 'The DID of the Identity whose Portfolio details are to be fetched',
+    type: 'string',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
+  })
+  @ApiParam({
+    name: 'portfolioId',
+    description: 'The portfolioId for which details are to be fetched',
```
Comment (prashantasdeveloper) [inline src/portfolios/portfolios.controller.ts:159]:
```suggestion
    description: 'The ID of the portfolio for which details are to be fetched. Use 0 for default Portfolio',
```

File: src/portfolios/portfolios.controller.ts [hunk 4]
```diff
@@ -143,4 +143,33 @@ export class PortfoliosController {
...
+    description: 'This endpoint will provide details for the provided Portfolio of an Identity',
+  })
+  @ApiParam({
+    name: 'did',
+    description: 'The DID of the Identity whose Portfolio details are to be fetched',
+    type: 'string',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
+  })
+  @ApiParam({
+    name: 'portfolioId',
```
Comment (prashantasdeveloper) [inline src/portfolios/portfolios.controller.ts:158]:
```suggestion
    name: 'id',
```

File: src/portfolios/portfolios.controller.ts (hunk 1 above)
Comment (prashantasdeveloper) [reply src/portfolios/portfolios.controller.ts:163]: Yes, this should be ApiOKResponse

File: src/portfolios/portfolios.controller.ts [hunk 5]
```diff
@@ -143,4 +143,33 @@ export class PortfoliosController {
...
+    description: 'The portfolioId for which details are to be fetched',
+    type: 'string',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
+  })
+  @ApiArrayResponse(PortfolioModel, {
+    description: 'Return the list of all Portfolios of the given Identity',
```
Comment (prashantasdeveloper) [inline src/portfolios/portfolios.controller.ts:164]:
```suggestion
    description: 'The details of the Portfolio',
```
You can also write `Portfolio details`

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]

**Review by prashantasdeveloper — APPROVED** [review]

**Pull request merged**